### ISSUE TITLE
Migrate our machines to the CAMH Research IPA server

### ISSUE THREAD

**Issue by pipitone** [TIGRLab/admin#446]
We have an IPA server we can use for authentication now. This will also make using the research storage much easier since user/group ids are supplied from IPA also. Doing this will require: 

 - we change all of our machine's domains to `camhres.ca`
 - use the camhres DNS server (which means we lose DNS lookups for the PET network.... shouldn't be much of an issue, except for email)
 - remove `nscd` before running `sssd` to configure IPA access. 

I will include the migration details in `/projects/admin/ipa-migration`, and create a few test VMs to test the migration.

**Comment by pipitone** [comment]
I've added a new role to ansible-playbooks called `ipa-client`. Assigning this role (and removing the `nis-client` role) should nicely convert a machine over to using the research IT IPA server for authentication. 
There is a vagrant test box in `roles/ipa-client/test` that you can use to demonstrate this, using the `ipa-migration.yml` playbook (see the README in `test/`)

The are only a few remaining issues before we can go live with with this: 
 - resolve a user name conflict with @edickie's user. Her user account is named the same on the SCC and in the tigrlab, so there is only the SCC account on the IPA server. The fix, as I see it, is to use the SCC account for our lab also (i can see us moving entirely over to SCC accounts eventually) by chown'ing all of edickie's files to the SCC UID.
 - test the IPA migration with SGE. I suspect there will be some hiccups here, but I haven't created a proper test environment yet to try it out. I'm inclined to say we just wing it... 
 - some of our private group IDs didn't transfer to IPA... I'm working on it, but it doesn't pose too big of an issue.

**Comment by pipitone** [comment]
I'm going to test the switch today on `kandel`. :8ball:

**Comment by pipitone** [comment]
Almost done;
- All machines identify themselves as `hostname.camhres.ca` and use the CAMH Research DNS server.
- I've reconciled all of our users and the SCC users (where people had two accounts, I switched the UID to match the SCC UIDs).
- I've rejigged the ansible playbooks to play nicely with the new hosts, and generally cleaned things up there a tad.

Remaining to do is:
- [ ] enroll `kimsrv` with IPA
- [ ] rename groups (`data` -> `kimel_data`, `admins` -> `kimel_admins`)
- [ ] destroy `srv-nis`
